telegram/_bot.py

```diff
@@ -310,6 +310,15 @@ def __init__(
         self._freeze()
 
     async def __aenter__(self: BT) -> BT:
+        """
+        |async_context_manager| :meth:`initializes <initialize>` the Bot.
+
+        Returns:
+            The initialized Bot instance.
+
+        Raises:
+            :exc:`Exception`: If an exception is raised during initialization.
```

Comment:
IMO it would be good to mention that `shutdown` is called in this case. similar for the other places :)